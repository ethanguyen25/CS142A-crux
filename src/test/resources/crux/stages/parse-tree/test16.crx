// shadow global with local

int a;

void main() {
  a = 1;

  int a;
  a = 2;
  printInt(a);
}
