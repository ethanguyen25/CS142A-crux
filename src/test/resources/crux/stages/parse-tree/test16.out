program
  declarationList
    declaration
      variableDeclaration
        type
    declaration
      functionDefinition
        type
        parameterList
        statementBlock
          statementList
            statement
              assignmentStatement
                designator
                expression0
                  expression1
                    expression2
                      expression3
                        literal
            statement
              variableDeclaration
                type
            statement
              assignmentStatement
                designator
                expression0
                  expression1
                    expression2
                      expression3
                        literal
            statement
              callStatement
                callExpression
                  expressionList
                    expression0
                      expression1
                        expression2
                          expression3
                            designator
